From 4ab87c2ff7f4d0bfad7762f9643bf5449dfa0e52 Mon Sep 17 00:00:00 2001
From: miwu04 <me@alanlin.icu>
Date: Sun, 21 Jun 2026 03:21:25 +0800
Subject: [PATCH 8/8] AOSCOS: Guard stray inclusion of Simulator-riscv64.h This
 is a leftover by Bug 2038968 D300034, but it should be fixed in Bug 2039343
 D300387 which happens to miss the Fx152 train.

Should be dropped by the release of Fx153.

Link: https://phabricator.services.mozilla.com/D300034 ("Bug 2038968 -
 Part 4: Make it an error to include simulator headers when simulator is
 disabled. r=jandem!")
Link: https://phabricator.services.mozilla.com/D300387 ("Bug 2039343 -
 Part 6: Remove InstructionGetters. r=jandem!")

Co-authored-by: CSharperMantle <rong.bao@csmantle.top>
---
 .../constant/Base-constant-riscv-cpp.patch      | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 src/js/src/jit/riscv64/constant/Base-constant-riscv-cpp.patch

diff --git a/src/js/src/jit/riscv64/constant/Base-constant-riscv-cpp.patch b/src/js/src/jit/riscv64/constant/Base-constant-riscv-cpp.patch
new file mode 100644
index 000000000..359cd343d
--- /dev/null
+++ b/src/js/src/jit/riscv64/constant/Base-constant-riscv-cpp.patch
@@ -0,0 +1,17 @@
+diff --git a/js/src/jit/riscv64/constant/Base-constant-riscv.cpp b/js/src/jit/riscv64/constant/Base-constant-riscv.cpp
+index b25ce174bc5b..40b7472ccec1 100644
+--- a/js/src/jit/riscv64/constant/Base-constant-riscv.cpp
++++ b/js/src/jit/riscv64/constant/Base-constant-riscv.cpp
+@@ -13,7 +13,11 @@
+ #include "jit/riscv64/constant/Constant-riscv-v.h"
+ #include "jit/riscv64/constant/Constant-riscv-zicsr.h"
+ #include "jit/riscv64/constant/Constant-riscv-zifencei.h"
+-#include "jit/riscv64/Simulator-riscv64.h"
++
++#ifdef JS_SIMULATOR_RISCV64
++#  include "jit/riscv64/Simulator-riscv64.h"
++#endif
++
+ namespace js {
+ namespace jit {
+ 
-- 
2.52.0

